### PR TITLE
ceph-backport.sh: implement --milestones feature and more-careful vetting

### PR DESCRIPTION
**Pull request by smithfarm**
This PR implements `ceph-backport.sh --milestones` which will find and fix PRs that are missing a milestone (or have the wrong milestone).

It also tries to be more careful in several areas:

* refuse to stage a backport if the backport tracker issue is closed
* check that the Backport tracker issue's "release" field is actually one of the active milestones on GitHub
* if `github_user` is not set, do not fall back to `$USER` (this leads to confusing behavior when the actual GitHub username is something other than `$USER`)
* even when `github_token` is set, do an actual GitHub API call and bail out if GitHub does not recognize the token
* refuse to cherry-pick from a PR that targets any branch other than `ceph:master`
* refuse to cherry-pick from a PR that isn't merged

### CODE REVIEW

**Comment by smithfarm** [comment]
@theanalyst @yuriw This PR implements `ceph-backport.sh --check-milestones` which will find and fix PRs that are missing a milestone (or have the wrong milestone).

**Comment by yuriw** [comment]
@smithfarm defer to @theanalyst

File: src/script/ceph-backport.sh [hunk 1]
```diff
@@ -77,8 +79,54 @@ function bail_out_github_api {
...
+        if [ "$tslc" = "$oks" ] ; then
+            debug "Tracker status $ts is OK for backport to proceed"
+            tslc_is_ok="yes"
+            break
+        fi
+    done
+    if [ "$tslc_is_ok" ] ; then
+        true
+    else
+        if [ "$tslc" = "in progress" ] ; then
```
Comment (theanalyst) [inline src/script/ceph-backport.sh]: There might be cases where someone took up a tracker issue, marked as in-progress and is stale now, how should these cases be handled?

Comment (smithfarm) [reply src/script/ceph-backport.sh]: `--force` option is coming in a follow-up PR

Comment (smithfarm) [reply src/script/ceph-backport.sh]: (Currently I'm handling that by manually editing the offending fields of the Backport tracker issue to stop the script from complaining.)

Comment (theanalyst) [reply src/script/ceph-backport.sh]: ah cool

**Pull request merged**